include/cvc5/cvc5_skolem_id.h

```diff
@@ -0,0 +1,333 @@
+   */
+  EVALUE(BAGS_MAP_SUM),
+  /** bag diff to witness (not (= A B)) */
+  EVALUE(BAGS_DEQ_DIFF),
+  /** Given a group term ((_ table.group n1 ... nk) A) of type (Bag (Table T))
+   * this uninterpreted function maps elements of A to their parts in the
+   * resulting partition. It has type (-> T (Table T))
+   */
+  EVALUE(TABLES_GROUP_PART),
+  /**
+   * Given a group term ((_ table.group n1 ... nk) A) of type (Bag (Table T))
+   * and a part B of type (Table T), this function returns a skolem element
+   * that is a member of B if B is not empty.
+   */
+  EVALUE(TABLES_GROUP_PART_ELEMENT),
+  /** Given a group term ((_ rel.group n1 ... nk) A) of type (Set (Relation T))
+   * this uninterpreted function maps elements of A to their parts in the
+   * resulting partition. It has type (-> T (Relation T))
+   */
+  EVALUE(RELATIONS_GROUP_PART),
+  /**
+   * Given a group term ((_ rel.group n1 ... nk) A) of type (Set (Relation T))
+   * and a part B of type (Relation T), this function returns a skolem element
+   * that is a member of B if B is not empty.
+   */
+  EVALUE(RELATIONS_GROUP_PART_ELEMENT),
+  /** An interpreted function for bag.choose operator:
+   * (choose A) is expanded as
+   * (witness ((x elementType))
+   *    (ite
+   *      (= A (as set.empty (Set E)))
+   *      (= x (uf A))
+   *      (and (set.member x A) (= x uf(A)))
+   * where uf: (Set E) -> E is a skolem function, and E is the type of elements
+   * of A
+   */
+  EVALUE(SETS_CHOOSE),
+  /** set diff to witness (not (= A B)) */
+  EVALUE(SETS_DEQ_DIFF),
+  EVALUE(SETS_FOLD_CARD),
+  EVALUE(SETS_FOLD_COMBINE),
+  EVALUE(SETS_FOLD_ELEMENTS),
+  EVALUE(SETS_FOLD_UNION),
+  /**
+   * A skolem variable that is unique per terms (set.map f A), y which is an
+   * element in (set.map f A). The skolem is constrained to be an element in A,
+   * and it is mapped to y by f.
+   */
+  EVALUE(SETS_MAP_DOWN_ELEMENT),
+  /** abstract value for a term t */
+  EVALUE(ABSTRACT_VALUE),
+  //================================================= Unknown rule
+  EVALUE(NONE),
+};
+// clang-format on
+
+#ifdef CVC5_API_USE_C_ENUMS
+#ifndef DOXYGEN_SKIP
+typedef enum ENUM(SkolemFunId) ENUM(SkolemFunId);
+#endif
+#endif
+
+#ifdef CVC5_API_USE_C_ENUMS
+
+/**
+ * Get a string representation of a Cvc5SkolemFunId.
+ * @param rule The proof rule.
+ * @return The string representation.
+ */
+const char* cvc5_skolem_id_to_string(Cvc5SkolemFunId kind);
+
+/**
+ * Hash function for Cvc5SkolemFunId.
+ * @param rule The proof rule.
+ * @return The hash value.
+ */
+size_t cvc5_skolem_id_hash(Cvc5SkolemFunId rule);
+
+#else
+/**
+ * Converts a proof rule to a string. Note: This function is also used in
+ * `safe_print()`. Changing this function name or signature will result in
+ * `safe_print()` printing "<unsupported>" instead of the proper strings for
+ * the enum values.
+ *
+ * @param id The proof rule
+ * @return The name of the proof rule
+ */
+const char* toString(SkolemFunId id);
```

Comment:
Use the new `std::to_string` convention in #10361.

Reply:
Done. Note we still require `toString` for `safe_print`. I've discussed this with @aniemetz  and we will refactor to eliminate the need for `const char * toString` methods throughout.